filebeat/crawler/prospector.go

```diff
@@ -197,7 +203,12 @@ func (p *Prospector) startHarvester(state input.FileState, offset int64) (*harve
 		return nil, err
 	}
 
-	h.Start()
+	go func() {
+		p.wg.Add(1)
```

Comment:
The `Add()` call should occur before the goroutine is started so just move it before the `go func()`.